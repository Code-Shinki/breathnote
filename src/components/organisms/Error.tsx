import React from 'react'
import styles from './Error.module.scss'

type Props = {
  status: number
  message: string
}

const ErrorArticle: React.FC<Props> = ({ status, message }) => {
  return (
    <article className={styles.article}>
      <h1 className={styles.title}>{status}</h1>
      <h2 className={styles.subtitle}>{message}</h2>
      <p className={styles.message}>申し訳ございません。サイト上でエラーが発生しました。</p>
      <p className={styles.message}>ホームへ戻るか、各種SNSから管理者へご連絡ください。</p>
    </article>
  )
}

export default ErrorArticle
