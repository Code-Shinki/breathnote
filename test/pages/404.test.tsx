import { cleanup, render, screen } from '@testing-library/react'
import React from 'react'
import Error from '../../src/pages/404'

describe(`404 (pages/404.tsx)`, () => {
  afterEach(() => {
    cleanup
  })

  test('snapshot', async () => {
    const option = {
      pageType: '404',
      fullPath: `https://blog.shinki.net/404`,
    }
    const { asFragment } = render(<Error option={option} />)
    const tree = asFragment()
    expect(tree).toMatchSnapshot()
  })

  test('integration', async () => {
    const option = {
      pageType: '404',
      fullPath: `https://blog.shinki.net/404`,
    }
    render(<Error option={option} />)

    // ヘッダーが存在する
    expect(screen.getByRole('banner')).toBeInTheDocument()
    // メインコンテンツが存在する
    expect(screen.getByRole('main')).toBeInTheDocument()
    // サイドコンテンツ (SHARE, CONTENTS, MORE) が存在する
    expect(screen.getAllByRole('complementary')).toHaveLength(3)
    // フッターが存在する
    expect(screen.getByRole('contentinfo')).toBeInTheDocument()

    // h1属性が1個だけ存在する
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument()
    // 記事が存在する
    expect(screen.getByRole('article')).toBeInTheDocument()
  })
})
