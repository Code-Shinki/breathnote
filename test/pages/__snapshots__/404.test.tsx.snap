// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 (pages/404.tsx) snapshot 1`] = `
<DocumentFragment>
  <article
    class="article"
  >
    <h1
      class="title"
    >
      404
    </h1>
    <h2
      class="subtitle"
    >
      Not Found.
    </h2>
    <p
      class="message"
    >
      申し訳ございません。サイト上でエラーが発生しました。
    </p>
    <p
      class="message"
    >
      ホームへ戻るか、各種SNSから管理者へご連絡ください。
    </p>
  </article>
</DocumentFragment>
`;
